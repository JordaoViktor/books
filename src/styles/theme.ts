export default {
  colors: {
    primary: "#FFFF",
    secondary: "#333333",
    tertiary: "#AB2680",
    background: "#E5E5E5",
    dark: "#000",

    lightenText: "#FFFF",
    darkText: "#333333",

    darkOpacity200: "rgba(0,0,0,0.20)",
    darkOpacity300: "rgba(0,0,0,0.32)",
    darkOpacity400: "rgba(0,0,0,0.4)",

    transparent: "transparent"
  },
  fonts: {
    light: "Heebo_300Light",
    normal: "Heebo_400Regular",
    bold: "Heebo_500Medium",
  },
}