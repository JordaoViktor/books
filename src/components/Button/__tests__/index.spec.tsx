import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import theme from '@styles/theme'
import { Button } from '../index';

describe('Button component', () => {

  it('should match a snapshot', () => {
    const FormInputRender = renderer
      .create(
        <ThemeProvider theme={theme} >
          <Button />
        </ThemeProvider>)
      .toJSON();

    expect(FormInputRender).toMatchSnapshot();
  })
});