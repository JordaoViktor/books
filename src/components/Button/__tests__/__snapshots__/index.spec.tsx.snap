// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match a snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFF",
      "borderColor": "transparent",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 36,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 9,
      "paddingLeft": 9,
      "paddingRight": 9,
      "paddingTop": 9,
    }
  }
/>
`;

exports[`matches button snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFF",
      "borderColor": "transparent",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 36,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 9,
      "paddingLeft": 9,
      "paddingRight": 9,
      "paddingTop": 9,
    }
  }
/>
`;
